2022-02-15 05:36:34 info: ---------------------------------------------
2022-02-15 05:36:34 info: >>> Running Unit Test `ex1-programmerFun` <<<
2022-02-15 05:36:34 info: ---------------------------------------------
2022-02-15 05:36:50 info: All unit tests passed.
2022-02-15 05:36:59 info: All steps were completed successfully
2022-02-15 08:05:40 info: ------------------------------------------
2022-02-15 08:05:40 info: >>> Running Unit Test `ex2-pokemonApp` <<<
2022-02-15 08:05:40 info: ------------------------------------------
2022-02-15 08:05:49 info: All unit tests passed.
2022-02-15 08:05:54 info: All steps were completed successfully
2022-02-15 08:18:39 info: -----------------------------------------
2022-02-15 08:18:39 info: >>> Running Unit Test `ex3-rollAnAce` <<<
2022-02-15 08:18:39 info: -----------------------------------------
2022-02-15 08:18:43 info: All unit tests passed.
2022-02-15 08:18:47 info: All steps were completed successfully
2022-02-15 08:37:45 info: ----------------------------------------
2022-02-15 08:37:45 info: >>> Running Unit Test `ex4-diceRace` <<<
2022-02-15 08:37:45 info: ----------------------------------------
2022-02-15 08:37:49 error: *** Unit Test Error Report ***

Command failed: npx jest ex4-diceRace.test.js --colors
 FAIL  3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js
  ex4-diceRace
    × should exist and be executable (3 ms)
    √ should have all TODO comments removed
    √ `rollDice` should not contain unneeded console.log calls (1 ms)
    × should use `dice.map()`
    × should use `Promise.race()`
    × should resolve as soon as a die settles successfully (1 ms)
    × should reject with an Error as soon as a die rolls off the table (1 ms)

  ● ex4-diceRace › should exist and be executable

    expect(received).toBeDefined()

    Received: undefined

      30 |
      31 |   test('should exist and be executable', () => {
    > 32 |     expect(exported).toBeDefined();
         |                      ^
      33 |   });
      34 |
      35 |   testTodosRemoved(() => source);

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:32:22)

  ● ex4-diceRace › should use `dice.map()`

    expect(received).toBeDefined()

    Received: undefined

      38 |
      39 |   test('should use `dice.map()`', () => {
    > 40 |     expect(state.diceMap).toBeDefined();
         |                           ^
      41 |   });
      42 |
      43 |   test('should use `Promise.race()`', () => {

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:40:27)

  ● ex4-diceRace › should use `Promise.race()`

    expect(received).toBeDefined()

    Received: undefined

      42 |
      43 |   test('should use `Promise.race()`', () => {
    > 44 |     expect(state.promiseAll).toBeDefined();
         |                              ^
      45 |   });
      46 |
      47 |   test('should resolve as soon as a die settles successfully', async () => {

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:44:30)

  ● ex4-diceRace › should resolve as soon as a die settles successfully

    expect(received).toBeDefined()

    Received: undefined

      47 |   test('should resolve as soon as a die settles successfully', async () => {
      48 |     expect.assertions(3);
    > 49 |     expect(exported).toBeDefined();
         |                      ^
      50 |
      51 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0);
      52 |

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:49:22)

  ● ex4-diceRace › should resolve as soon as a die settles successfully

    expect.assertions(3)

    Expected three assertions to be called but received one assertion call.

      46 |
      47 |   test('should resolve as soon as a die settles successfully', async () => {
    > 48 |     expect.assertions(3);
         |            ^
      49 |     expect(exported).toBeDefined();
      50 |
      51 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0);

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:48:12)

  ● ex4-diceRace › should reject with an Error as soon as a die rolls off the table

    expect(received).toBeDefined()

    Received: undefined

      63 |   test('should reject with an Error as soon as a die rolls off the table', async () => {
      64 |     expect.assertions(3);
    > 65 |     expect(exported).toBeDefined();
         |                      ^
      66 |
      67 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0.999);
      68 |

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:65:22)

  ● ex4-diceRace › should reject with an Error as soon as a die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received one assertion call.

      62 |
      63 |   test('should reject with an Error as soon as a die rolls off the table', async () => {
    > 64 |     expect.assertions(3);
         |            ^
      65 |     expect(exported).toBeDefined();
      66 |
      67 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0.999);

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:64:12)

Test Suites: 1 failed, 1 total
Tests:       5 failed, 2 passed, 7 total
Snapshots:   0 total
Time:        1.061 s
Ran all test suites matching /ex4-diceRace.test.js/i.
2022-02-15 08:37:51 error: *** ESLint Report ***

ex4-diceRace.js
  38:1  error  Parsing error: Unexpected token

✖ 1 problem (1 error, 0 warnings)


2022-02-15 08:39:54 info: ----------------------------------------
2022-02-15 08:39:54 info: >>> Running Unit Test `ex4-diceRace` <<<
2022-02-15 08:39:54 info: ----------------------------------------
2022-02-15 08:39:58 error: *** Unit Test Error Report ***

Command failed: npx jest ex4-diceRace.test.js --colors
 FAIL  3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js
  ex4-diceRace
    × should exist and be executable (3 ms)
    √ should have all TODO comments removed
    √ `rollDice` should not contain unneeded console.log calls (1 ms)
    × should use `dice.map()` (1 ms)
    × should use `Promise.race()` (1 ms)
    × should resolve as soon as a die settles successfully
    × should reject with an Error as soon as a die rolls off the table (1 ms)

  ● ex4-diceRace › should exist and be executable

    expect(received).toBeDefined()

    Received: undefined

      30 |
      31 |   test('should exist and be executable', () => {
    > 32 |     expect(exported).toBeDefined();
         |                      ^
      33 |   });
      34 |
      35 |   testTodosRemoved(() => source);

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:32:22)

  ● ex4-diceRace › should use `dice.map()`

    expect(received).toBeDefined()

    Received: undefined

      38 |
      39 |   test('should use `dice.map()`', () => {
    > 40 |     expect(state.diceMap).toBeDefined();
         |                           ^
      41 |   });
      42 |
      43 |   test('should use `Promise.race()`', () => {

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:40:27)

  ● ex4-diceRace › should use `Promise.race()`

    expect(received).toBeDefined()

    Received: undefined

      42 |
      43 |   test('should use `Promise.race()`', () => {
    > 44 |     expect(state.promiseAll).toBeDefined();
         |                              ^
      45 |   });
      46 |
      47 |   test('should resolve as soon as a die settles successfully', async () => {

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:44:30)

  ● ex4-diceRace › should resolve as soon as a die settles successfully

    expect(received).toBeDefined()

    Received: undefined

      47 |   test('should resolve as soon as a die settles successfully', async () => {
      48 |     expect.assertions(3);
    > 49 |     expect(exported).toBeDefined();
         |                      ^
      50 |
      51 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0);
      52 |

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:49:22)

  ● ex4-diceRace › should resolve as soon as a die settles successfully

    expect.assertions(3)

    Expected three assertions to be called but received one assertion call.

      46 |
      47 |   test('should resolve as soon as a die settles successfully', async () => {
    > 48 |     expect.assertions(3);
         |            ^
      49 |     expect(exported).toBeDefined();
      50 |
      51 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0);

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:48:12)

  ● ex4-diceRace › should reject with an Error as soon as a die rolls off the table

    expect(received).toBeDefined()

    Received: undefined

      63 |   test('should reject with an Error as soon as a die rolls off the table', async () => {
      64 |     expect.assertions(3);
    > 65 |     expect(exported).toBeDefined();
         |                      ^
      66 |
      67 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0.999);
      68 |

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:65:22)

  ● ex4-diceRace › should reject with an Error as soon as a die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received one assertion call.

      62 |
      63 |   test('should reject with an Error as soon as a die rolls off the table', async () => {
    > 64 |     expect.assertions(3);
         |            ^
      65 |     expect(exported).toBeDefined();
      66 |
      67 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0.999);

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:64:12)

Test Suites: 1 failed, 1 total
Tests:       5 failed, 2 passed, 7 total
Snapshots:   0 total
Time:        1.227 s
Ran all test suites matching /ex4-diceRace.test.js/i.
2022-02-15 08:40:00 error: *** ESLint Report ***

ex4-diceRace.js
  38:1  error  Parsing error: Unexpected token

✖ 1 problem (1 error, 0 warnings)


2022-02-15 08:41:34 info: ----------------------------------------
2022-02-15 08:41:34 info: >>> Running Unit Test `ex4-diceRace` <<<
2022-02-15 08:41:34 info: ----------------------------------------
2022-02-15 08:41:38 error: *** Unit Test Error Report ***

Command failed: npx jest ex4-diceRace.test.js --colors
 FAIL  3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js
  ex4-diceRace
    × should exist and be executable (5 ms)
    √ should have all TODO comments removed (1 ms)
    √ `rollDice` should not contain unneeded console.log calls (1 ms)
    × should use `dice.map()`
    × should use `Promise.race()` (1 ms)
    × should resolve as soon as a die settles successfully (1 ms)
    × should reject with an Error as soon as a die rolls off the table (1 ms)

  ● ex4-diceRace › should exist and be executable

    expect(received).toBeDefined()

    Received: undefined

      30 |
      31 |   test('should exist and be executable', () => {
    > 32 |     expect(exported).toBeDefined();
         |                      ^
      33 |   });
      34 |
      35 |   testTodosRemoved(() => source);

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:32:22)

  ● ex4-diceRace › should use `dice.map()`

    expect(received).toBeDefined()

    Received: undefined

      38 |
      39 |   test('should use `dice.map()`', () => {
    > 40 |     expect(state.diceMap).toBeDefined();
         |                           ^
      41 |   });
      42 |
      43 |   test('should use `Promise.race()`', () => {

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:40:27)

  ● ex4-diceRace › should use `Promise.race()`

    expect(received).toBeDefined()

    Received: undefined

      42 |
      43 |   test('should use `Promise.race()`', () => {
    > 44 |     expect(state.promiseAll).toBeDefined();
         |                              ^
      45 |   });
      46 |
      47 |   test('should resolve as soon as a die settles successfully', async () => {

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:44:30)

  ● ex4-diceRace › should resolve as soon as a die settles successfully

    expect(received).toBeDefined()

    Received: undefined

      47 |   test('should resolve as soon as a die settles successfully', async () => {
      48 |     expect.assertions(3);
    > 49 |     expect(exported).toBeDefined();
         |                      ^
      50 |
      51 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0);
      52 |

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:49:22)

  ● ex4-diceRace › should resolve as soon as a die settles successfully

    expect.assertions(3)

    Expected three assertions to be called but received one assertion call.

      46 |
      47 |   test('should resolve as soon as a die settles successfully', async () => {
    > 48 |     expect.assertions(3);
         |            ^
      49 |     expect(exported).toBeDefined();
      50 |
      51 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0);

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:48:12)

  ● ex4-diceRace › should reject with an Error as soon as a die rolls off the table

    expect(received).toBeDefined()

    Received: undefined

      63 |   test('should reject with an Error as soon as a die rolls off the table', async () => {
      64 |     expect.assertions(3);
    > 65 |     expect(exported).toBeDefined();
         |                      ^
      66 |
      67 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0.999);
      68 |

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:65:22)

  ● ex4-diceRace › should reject with an Error as soon as a die rolls off the table

    expect.assertions(3)

    Expected three assertions to be called but received one assertion call.

      62 |
      63 |   test('should reject with an Error as soon as a die rolls off the table', async () => {
    > 64 |     expect.assertions(3);
         |            ^
      65 |     expect(exported).toBeDefined();
      66 |
      67 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0.999);

      at Object.<anonymous> (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:64:12)

Test Suites: 1 failed, 1 total
Tests:       5 failed, 2 passed, 7 total
Snapshots:   0 total
Time:        1.289 s, estimated 2 s
Ran all test suites matching /ex4-diceRace.test.js/i.
2022-02-15 08:41:40 error: *** ESLint Report ***

ex4-diceRace.js
  38:1  error  Parsing error: Unexpected token

✖ 1 problem (1 error, 0 warnings)


2022-02-15 08:44:31 info: ----------------------------------------
2022-02-15 08:44:31 info: >>> Running Unit Test `ex4-diceRace` <<<
2022-02-15 08:44:31 info: ----------------------------------------
2022-02-15 08:44:35 info: All unit tests passed.
2022-02-15 08:44:40 info: All steps were completed successfully
2022-02-15 08:46:09 info: ----------------------------------------
2022-02-15 08:46:09 info: >>> Running Unit Test `ex4-diceRace` <<<
2022-02-15 08:46:09 info: ----------------------------------------
2022-02-15 08:46:13 info: All unit tests passed.
2022-02-15 08:46:17 info: All steps were completed successfully
2022-02-15 08:54:05 info: --------------------------------------------
2022-02-15 08:54:05 info: >>> Running Unit Test `ex6-browserDebug` <<<
2022-02-15 08:54:05 info: --------------------------------------------
2022-02-15 08:54:05 warn: A unit test file was not provided.
2022-02-15 08:54:07 error: *** ESLint Report ***

ex6-browserDebug/index.js
  65:1  error  Parsing error: Unexpected token

✖ 1 problem (1 error, 0 warnings)


2022-02-15 08:55:46 info: --------------------------------------------
2022-02-15 08:55:46 info: >>> Running Unit Test `ex6-browserDebug` <<<
2022-02-15 08:55:46 info: --------------------------------------------
2022-02-15 08:55:46 warn: A unit test file was not provided.
2022-02-15 08:55:50 info: All steps were completed successfully
2022-02-15 08:58:22 info: ----------------------------------------
2022-02-15 08:58:22 info: >>> Running Unit Test `ex5-vscDebug` <<<
2022-02-15 08:58:22 info: ----------------------------------------
2022-02-15 08:58:22 warn: A unit test file was not provided.
2022-02-15 08:58:27 info: All steps were completed successfully
2022-02-19 03:23:17 info: ---------------------------------------------
2022-02-19 03:23:17 info: >>> Running Unit Test `ex1-programmerFun` <<<
2022-02-19 03:23:17 info: ---------------------------------------------
2022-02-19 03:23:33 info: All unit tests passed.
2022-02-19 03:23:39 info: All steps were completed successfully
2022-02-19 06:27:10 info: ------------------------------------------
2022-02-19 06:27:10 info: >>> Running Unit Test `ex2-pokemonApp` <<<
2022-02-19 06:27:10 info: ------------------------------------------
2022-02-19 06:27:16 info: All unit tests passed.
2022-02-19 06:27:17 error: *** ESLint Report ***

ex2-pokemonApp/index.js
  32:5  warning  Please do not use `innerHTML` in the homework. Use `textContent` and/or `document.createElement()` instead  no-restricted-syntax

✖ 1 problem (0 errors, 1 warning)


2022-02-19 06:37:36 info: ------------------------------------------
2022-02-19 06:37:36 info: >>> Running Unit Test `ex2-pokemonApp` <<<
2022-02-19 06:37:36 info: ------------------------------------------
2022-02-19 06:37:42 info: All unit tests passed.
2022-02-19 06:37:46 info: All steps were completed successfully
